Implement the BooleanFilter

chervaliery opened the pull request:
In order to make query on boolean type, I implement the BooleanFilter.

greg0ire commented on Filter/BooleanFilter.php:
```diff
@@ -32,23 +32,35 @@ public function filter(ProxyQueryInterface $queryBuilder, $alias, $field, $data)
                 }
 
-                $values[] = ($v == BooleanType::TYPE_YES) ? 1 : 0;
+                $values[] = ($v == BooleanType::TYPE_YES) ? true : false;
```
Could you change that to simply : `$values[] = ($v == BooleanType::TYPE_YES)` ? It gives the same result.

chervaliery replied:
It is not the same, 
BooleanType::TYPE_YES = 1 and BooleanType::TYPE_YES = 2
https://github.com/sonata-project/SonataCoreBundle/blob/master/Form/Type/BooleanType.php

greg0ire replied:
`($v == BooleanType::TYPE_YES)` is a condition. It always returns a boolean (either `true` or `false`). This line is basically saying : If the condition is `true`, store `true`, if it is `false` store `false`. I'm saying we could replace that with just "store the result of the condition"

greg0ire commented on Filter/BooleanFilter.php:
```diff
@@ -32,23 +32,35 @@ public function filter(ProxyQueryInterface $queryBuilder, $alias, $field, $data)
+            $queryBuilder
+            ->fieldOpen('terms')
+            ->field($field, $values)
+            ->fieldClose();
```
Can you indent these three lines ?

greg0ire commented on Filter/BooleanFilter.php:
```diff
@@ -32,23 +32,35 @@ public function filter(ProxyQueryInterface $queryBuilder, $alias, $field, $data)
+            $queryBuilder
+            ->fieldOpen('term')
+            ->field($field, ($data['value'] == BooleanType::TYPE_YES) ? true : false)
+            ->fieldClose();
```
Indention please

greg0ire commented on Filter/BooleanFilter.php:
```diff
@@ -32,23 +32,35 @@ public function filter(ProxyQueryInterface $queryBuilder, $alias, $field, $data)
+            $queryBuilder = new \Elastica\Query\Builder();
+            $queryBuilder
+            ->fieldOpen('term')
+            ->field($field, ($data['value'] == BooleanType::TYPE_YES) ? true : false)
```
Please remove this ternary.

greg0ire commented:
Could you please write some tests ? #7 introduces one for the `StringFilter`

greg0ire commented:
Your commit message is great! If you can fix all the things I pointed out, this will be merged quickly. Thanks a lot!

chervaliery committed:
Implement the BooleanFilter

This implementation allows to query on Boolean type e.g.:
{ "term": { "published": true }}

add some tests on the BooleanFilter

chervaliery commented:
Is there still a problem? Do I need to change something else?

greg0ire commented:
Huh yes there is a problem : there are 3 commits instead of one O_o

greg0ire commented on Tests/Filter/BooleanFilterTest.php:
```diff
@@ -0,0 +1,59 @@
+		$this->assertEquals('true', $queryArray['query']['bool']['must'][0]['term']['foo']);
+	}
+
+}
```
Please don't use tabs. Make sure you don't violate PSR. Try to integrate [this](http://cs.sensiolabs.org/) in your IDE if you can.

greg0ire commented:
I noticed that the commit have the same subject, but not the same body. The commit subject is not a category and the body a category. I think that in fact, you can keep the 3 commits, but they should be reworded.

greg0ire commented:
The first one is good.

greg0ire commented:
The second one should read like this : 

> add tests for the boolean filter

but it could also be squashed with the first commit, your choice.

greg0ire commented:
The third should read 

> remove unnecessary ternary
> 
> We want boolean here, not integers.

It must be squashed with the first commit if and the first commit does not work without it, otherwise you can keep it apart.

chervaliery commented:
Ok, I will squash them and change the tabs

greg0ire commented:
> Ok, I will squash them

I think this will be more simple :wink:

chervaliery commented:
This time it should be good !

greg0ire commented on Resources/config/services.xml:
```diff
@@ -44,5 +44,10 @@
+        <service
+            id="sonata.admin.search.filter.type.boolean"
+            class="Sonata\AdminSearchBundle\Filter\BooleanFilter">
+            <tag name="sonata.admin.filter.type" alias="elastica_boolean" />
```
Can you rebase on master and realias this service appropriately ?

chervaliery replied:
I did it

greg0ire replied:
Nice! Let's wait for the build to finish.

greg0ire commented:
Merged! Thanks a lot!
